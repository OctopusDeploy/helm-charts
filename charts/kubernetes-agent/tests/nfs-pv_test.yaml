suite: "persistence"
templates:
- templates/nfs-pv.yaml
tests:
- it: "should match snapshot"
  asserts:
  - matchSnapshot: {}

- it: "is not created when storageClassName has a value"
  set:
    persistence:
      storageClassName: "value"
  asserts:
  - hasDocuments:
      count: 0

- it: "is created when storageClassName is empty"
  set:
    persistence:
      storageClassName: ""
  asserts:
  - hasDocuments:
      count: 1

- it: "has Helm Release name in name"
  set:
    persistence:
      storageClassName: ""
  asserts:
  - matchRegex:
      path: metadata.name
      pattern: "release-name"

- it: "set storage capacity to be persistence size"
  set:
    persistence:
      size: 100Gi
      storageClassName: ""
  asserts:
  - equal:
      path: spec.capacity.storage
      value: 100Gi

- it: "uses NFS service DNS as server address"
  set:
    persistence:
      storageClassName: ""
  asserts:
  - equal:
      path: spec.csi.volumeAttributes.server
      value: octopus-agent-nfs.NAMESPACE.svc.cluster.local
  - equal:
      path: spec.csi.volumeAttributes.share
      value: "/"

