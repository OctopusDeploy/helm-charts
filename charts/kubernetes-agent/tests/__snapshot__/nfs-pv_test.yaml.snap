should match snapshot:
  1: |
    apiVersion: v1
    kind: PersistentVolume
    metadata:
      annotations:
        pv.kubernetes.io/provisioned-by: nfs.csi.k8s.io
      name: octopus-agent-nfs-release-name-pv-10gi
    spec:
      accessModes:
        - ReadWriteMany
      capacity:
        storage: 10Gi
      csi:
        driver: nfs.csi.k8s.io
        volumeAttributes:
          server: octopus-agent-nfs.NAMESPACE.svc.cluster.local
          share: /
        volumeHandle: octopus-agent-nfs.NAMESPACE.svc.cluster.local/octopus##
      mountOptions:
        - nfsvers=4.1
        - lookupcache=none
        - soft
        - timeo=50
        - retrans=4
      persistentVolumeReclaimPolicy: Retain
      storageClassName: octopus-agent-nfs-RELEASE-NAME-csi
